src/vs/platform/telemetry/common/fileTelemetryService.ts

```diff
@@ -0,0 +1,65 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { ITelemetryService, ITelemetryInfo, ITelemetryData } from 'vs/platform/telemetry/common/telemetry';
+import * as fs from 'fs';
+import { O_CREAT, O_WRONLY } from 'constants';
+import { Disposable } from 'vs/base/common/lifecycle';
+
+/**
+ * Write telemetry into a file for test purposes
+ */
+export class FileTelemetryService extends Disposable implements ITelemetryService {
+	_serviceBrand: undefined;
```

Comment:
Since this doesn't exist in the vscode repo anymore how about marking with with a SQL CARBON EDIT so it's clear we're adding this back in on purpose?

Reply:
👍 on this. Not sure if we should also just move to the SQL side of things now? That would avoid conflicts where we try to delete this?

Reply:
That's an even better idea (if Anthony's question about using existing functionality doesn't end up working out)